src/app/ApplicationTemplate.Tests/Presentation/NavigationTestsBase.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using ApplicationTemplate.Presentation;
+using Chinook.DynamicMvvm;
+using Chinook.SectionsNavigation;
+using Chinook.StackNavigation;
+using Xunit;
+
+namespace ApplicationTemplate.Tests
+{
+	public class NavigationTestsBase : TestBase
+	{
+		/// <summary>
+		/// Navigates from the <typeparamref name="TSourceViewModel"/> to the <typeparamref name="TSourceViewModel"/>
+		/// by executing the specified <paramref name="navigationCommand"/>.
+		/// </summary>
+		/// <typeparam name="TSourceViewModel">Type of the source ViewModel</typeparam>
+		/// <typeparam name="TDestinationViewModel">Type of the destination ViewModel</typeparam>
+		/// <param name="navigationCommand">Navigating command to </param>
+		/// <returns>Destination ViewModel</returns>
+		protected async Task<TDestinationViewModel> ExecuteNavigation<TSourceViewModel, TDestinationViewModel>(Func<TSourceViewModel, IDynamicCommand> navigationCommand)
```

Comment:
I think I would rename this considering that the method asserts. (It's not just navigating)

Reply:
Actually this method is barely useful. I actually moved it into the appropriate test file.